tests/integration/test_css_select_by_id.py

```diff
@@ -0,0 +1,45 @@
+from __future__ import unicode_literals
+import pytest
+
+import dash
+import dash_html_components as html
+from dash._utils import css_escape
+
+
+TEST_STRING_IDS = [
+    "standard-sane-id",
+    "!sane~ID:at>all?" "-",
+    "-1a",
+    "0abc",
+    "{'a': 1, 'b': 2, 'c': [3, 4, 5]}",
+    '"string-in-string-with-escape-characters"',
+    "\u001fabc\u007f",
+]
+
+TEST_DICT_IDS = [
+    {"a": 1, "b": "some-string", "c": False},
+    {"type": "some-type", "index": 42},
+]
+
+
+@pytest.mark.parametrize("id", TEST_STRING_IDS)
+def test_css_escape(dash_duo, id):
+    assert dash_duo.driver.execute_script(
+        "return CSS.escape({!r})".format(id)
+    ) == css_escape(id)
+
+
+@pytest.mark.parametrize(
+    "id", TEST_STRING_IDS + TEST_DICT_IDS,
+)
+def test_found_by_css_selector(dash_duo, id):
+    # This test also indirectly checks that output from the JavaScript stringify_id
+    # function equals output from the corresponding Python function.
+    app = dash.Dash(__name__)
+
+    app.layout = html.Div(id=id)
+
+    dash_duo.start_server(app)
+
+    assert dash_duo.wait_for_element_by_id(dash.stringify_id(id, escape_css=True))
```

Comment:
There is an easy possibility of extending
https://github.com/plotly/dash/blob/e8ac94919105a91c76a966c21aca2ec7b0297e22/dash/testing/browser.py#L277-L288
to check if input `element_id` is dict, and then do `dash.stringify_id(element_id, escape_css=True)` for the user in the `wait_for_element_by_id` function.

The user's test line here
```python
assert dash_duo.wait_for_element_by_id(dash.stringify_id(id, escape_css=True))
```
then becomes
```python
assert dash_duo.wait_for_element_by_id(id)
```
both in the case with string IDs, and dictionary IDs (when you use pattern-matching callbacks).

Not sure if that is too magical :crystal_ball: or a nice and easy to understand convenience.

Reply:
Good idea with extending the testing utilities, I think that is where most cases would be for escaping.